pkg/cloud/amazon/storage/bucket_provider.go

```diff
@@ -22,12 +26,11 @@ import (
 type AmazonBucketProvider struct {
 	Requirements *config.RequirementsConfig
 	api          s3iface.S3API
+	uploader     s3manageriface.UploaderAPI
+	downloader   s3manageriface.DownloaderAPI
 }
 
-func (b *AmazonBucketProvider) s3() (s3iface.S3API, error) {
-	if b.api != nil {
-		return b.api, nil
-	}
+func (b AmazonBucketProvider) createAWSSession() (*session.Session, error) {
```

Comment:
nit: missing `*` from `func (b *AmazonBucketProvider)`

Reply:
That's actually fine, don't need to modify anything from the struct in that function.

Reply:
Yeah, it's fine because the compiler will take care of and covert it into a pointer when it is required, but usually linters do complain about this inconsistency.